pyo3-derive-backend/src/pyclass.rs

```diff
@@ -384,9 +382,23 @@ fn impl_class(
             const FLAGS: usize = #(#flags)|* | #extended;
 
             #[inline]
-            unsafe fn type_object() -> &'static mut pyo3::ffi::PyTypeObject {
-                static mut TYPE_OBJECT: pyo3::ffi::PyTypeObject = pyo3::ffi::PyTypeObject_INIT;
-                &mut TYPE_OBJECT
+            fn type_object() -> std::ptr::NonNull<pyo3::ffi::PyTypeObject> {
```

Comment:
How about moving this code to impl block of `LazyTypeObject`?
Then we have 
```rust
impl LazyTypeObject {
...
    pub fn get_pyclass_type<T: PyClass>(&self) -> NonNull<ffi::PyTypeObject> { ... }
}

Reply:
Thanks, that's a really nice idea! It makes the macro code a lot neater 😄